### PR TITLE
refactor: prune amm related functions and types

### PR DESCRIPTION
**Pull request by yutianwu**
### Description

This pr will prune all the amm related functions and types.

### Changes

Notable changes:
* prun amm related functions and types

### CODE REVIEW

File: CONTRACTS_DESIGN.md [hunk 1]
```diff
@@ -108,28 +106,6 @@ Deposit BEP20 token to L2, `_accountName` will receive the token. This function
 
 Due to a ceiling on the code size of `ZkBNB` contract, `AdditionalZkBNB` will store more logic code which could not be stored on `ZkBNB`.
```
Comment (unclezoro) [inline CONTRACTS_DESIGN.md:107]: There still have some testing code like `createPair`, please delete them.

File: contracts/AssetGovernance.sol [hunk 2]
```diff
@@ -53,34 +44,19 @@ contract AssetGovernance is ReentrancyGuard {
     /// @notice Address that collects listing payments
     address public treasury;
 
-    /// @notice AccountIndex that collects listing payments
-    uint32 public treasuryAccountIndex;
```
Comment (unclezoro) [inline contracts/AssetGovernance.sol:57]: why `treasuryAccountIndex` is deleted? according to the comments, it is not related with AMM swap.

File: contracts/Governance.sol [hunk 3]
```diff
@@ -127,18 +127,4 @@ contract Governance is Config {
         require(!pausedAssets[assetId], "2i");
         return assetId;
     }
-
```
Comment (guagualvcha) [inline contracts/Governance.sol:130]: Does this related with AMM swap?

File: contracts/ZkBNB.sol [hunk 4]
```diff
@@ -250,7 +238,7 @@ contract ZkBNB is UpgradeableMaster, Events, Storage, Config, ReentrancyGuardUpg
         uint256 balanceAfter = _token.balanceOf(address(this));
         uint128 depositAmount = SafeCast.toUint128(balanceAfter.sub(balanceBefore));
         require(depositAmount <= MAX_DEPOSIT_AMOUNT, "C");
-        require(depositAmount > 0, "D");
+        //        require(depositAmount > 0, "D");
```
Comment (keefel) [inline contracts/ZkBNB.sol]: Why comment this?

File: contracts/ZkBNB.sol [hunk 5]
```diff
@@ -261,15 +249,77 @@ contract ZkBNB is UpgradeableMaster, Events, Storage, Config, ReentrancyGuardUpg
...
+        // check if the nft is mint from layer-2
+        bytes32 nftKey = keccak256(abi.encode(_nftL1Address, _nftL1TokenId));
+        uint16 collectionId = 2**16-1;
+        uint40 nftIndex = 2**40-1;
+        uint32 creatorAccountIndex = 2**32-1;
+        uint16 creatorTreasuryRate = 0;
+        bytes32 nftContentHash;
+        uint8 isNewNft = 1;
+        if (l2Nfts[nftKey].nftContentHash == bytes32(0)) {
+            // it means this is a new layer-1 nft
```
Comment (keefel) [inline contracts/ZkBNB.sol:279]: We have the conclusion not to deposit layer-1 nft?

File: contracts/ZkBNB.sol [hunk 6]
```diff
@@ -261,15 +249,77 @@ contract ZkBNB is UpgradeableMaster, Events, Storage, Config, ReentrancyGuardUpg
...
+        bytes memory pubData = TxTypes.writeDepositNftPubDataForPriorityQueue(_tx);
+
+        // add into priority request queue
+        addPriorityRequest(TxTypes.TxType.DepositNft, pubData);
+
+        emit DepositNft(accountNameHash, nftContentHash, _nftL1Address, _nftL1TokenId, collectionId);
     }
 
     function withdrawOrStoreNFT(TxTypes.WithdrawNft memory op) internal {
+        require(op.nftIndex < 2**40-1, "invalid nft index");
```
Comment (keefel) [inline contracts/ZkBNB.sol]: Defining 2**40-1 as a constant may be better.

**Review by unclezoro — APPROVED** [review]

File: contracts/ZkBNB.sol (hunk 5 above)
Comment (yutianwu) [reply contracts/ZkBNB.sol:279]: we don't do this in this pr

**Pull request merged**